src/components/cc-header-app/cc-header-app.js

```diff
@@ -150,148 +174,156 @@ export class CcHeaderApp extends LitElement {
     return i18n('cc-header-app.state-msg.unknown-state');
   };
 
-  _getLastUserActionMsg () {
-    if (this._lastUserAction === 'start') {
-      return i18n('cc-header-app.user-action-msg.app-will-start');
-    }
-    if (this._lastUserAction === 'restart') {
-      return i18n('cc-header-app.user-action-msg.deploy-will-begin');
-    }
-    if (this._lastUserAction === 'cancel') {
-      return i18n('cc-header-app.user-action-msg.deploy-cancelled');
-    }
-    if (this._lastUserAction === 'stop') {
-      return i18n('cc-header-app.user-action-msg.app-will-stop');
-    }
-  };
-
-  _onStart (type) {
-    this._lastUserAction = 'start';
-    dispatchCustomEvent(this, 'start', type);
+  /** @private */
+  _onCancel () {
+    this._lastUserAction = 'cancel';
+    dispatchCustomEvent(this, 'cancel');
   }
 
+  /**
+   * @param {'normal'|'rebuild'|'last-commit'} type
+   * @private
+   */
   _onRestart (type) {
     this._lastUserAction = 'restart';
     dispatchCustomEvent(this, 'restart', type);
   }
 
-  _onCancel () {
-    this._lastUserAction = 'cancel';
-    dispatchCustomEvent(this, 'cancel');
+  /**
+   * @param {'normal'|'rebuild'|'last-commit'} type
+   * @private
+   */
+  _onStart (type) {
+    this._lastUserAction = 'start';
+    dispatchCustomEvent(this, 'start', type);
   }
 
+  /** @private */
   _onStop () {
     this._lastUserAction = 'stop';
     dispatchCustomEvent(this, 'stop');
   }
 
-  willUpdate (changeProperties) {
-    if (changeProperties.has('status')) {
-      this._lastUserAction = (changeProperties.get('status') !== this.status)
-        ? null
-        : this._lastUserAction;
+  /**
+   * @param {CcHeaderAppChangedProperties} changedProperties
+   */
+  willUpdate (changedProperties) {
+    if (changedProperties.has('statusAndCommitState') && this.statusAndCommitState.type === 'loading') {
+      this._lastUserAction = null;
     }
   }
```

Comment:
@hsablonniere I'm not sure I understood the point of this one.

I thought the goal was to reset `_lastUserAction` but I'm still not sure, I feel like I'm missing something :thinking: